From a13dfd591ce814e8156ec400ac7b725e9d98e676 Mon Sep 17 00:00:00 2001
From: Joel Galenson <jgalenson@google.com>
Date: Wed, 17 Jul 2019 08:38:42 -0700
Subject: [PATCH] Hackily disable the __FILE__ macro in compiler-rt.

The use of the __FILE__ macro in compiler-rt stops it and rustc's
libcompiler_builtins from building reproducibly across directories.
As a hack, remove the macro.

Note that this is definitely a proof of concept hack.  A better
solution would use a flag like NDEBUG or use
https://reviews.llvm.org/D49466 to remap the paths used by the macro.
---
 compiler-rt/lib/builtins/int_util.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler-rt/lib/builtins/int_util.h b/compiler-rt/lib/builtins/int_util.h
index c3c87381ad8..40954460969 100644
--- a/compiler-rt/lib/builtins/int_util.h
+++ b/compiler-rt/lib/builtins/int_util.h
@@ -20,7 +20,7 @@
 #define INT_UTIL_H
 
 /** \brief Trigger a program abort (or panic for kernel code). */
-#define compilerrt_abort() __compilerrt_abort_impl(__FILE__, __LINE__, __func__)
+#define compilerrt_abort() __compilerrt_abort_impl(0, __LINE__, __func__)
 
 NORETURN void __compilerrt_abort_impl(const char *file, int line,
                                       const char *function);
-- 
2.22.0.510.g264f2c817a-goog

